R/dataframe-frame.R

```diff
@@ -112,19 +113,57 @@ dataframe__to_struct <- function(name = "") {
     wrap()
 }
 
+#' Convert an existing DataFrame to a LazyFrame
+#' @description Start a new lazy query from a DataFrame.
+#'
+#' @inherit as_polars_lf return
+#' @examples
+#' pl$DataFrame(a = 1:2, b = c(NA, "a"))$lazy()
 dataframe__lazy <- function() {
   self$`_df`$lazy() |>
     wrap()
 }
 
+#' Clone a DataFrame
+#'
+#' This is a cheap operation that does not copy data. It is rarely useful as
+#' `DataFrame`s are nearly 100% immutable, meaning that any modification of a
+#' `DataFrame` would lead to a clone anyways, but this can be useful when
+#' dealing with attributes (see examples).
+#'
+#' @inherit as_polars_df return
+#' @examples
+#' df1 <- as_polars_df(iris)
+#'
+#' # Make a function to take a DataFrame, add an attribute, and return a DataFrame
+#' give_attr <- function(data) {
+#'   attr(data, "created_on") <- "2024-01-29"
+#'   data
+#' }
+#' df2 <- give_attr(df1)
+#'
+#' # Problem: the original DataFrame also gets the attribute while it shouldn't!
+#' attributes(df1)
+#'
+#' # Use $clone() inside the function to avoid that
+#' give_attr <- function(data) {
+#'   data <- data$clone()
+#'   attr(data, "created_on") <- "2024-01-29"
+#'   data
+#' }
+#' df1 <- as_polars_df(iris)
+#' df2 <- give_attr(df1)
+#'
+#' # now, the original DataFrame doesn't get this attribute
+#' attributes(df1)
 dataframe__clone <- function() {
   self$`_df`$clone() |>
     wrap()
 }
 
-#' Get the DataFrame as a List of Series
+#' Get the DataFrame as a list of Series
 #'
-#' @return A [list] of [Series]
+#' @return A list of [Series]
```

Comment:
Why you edit this?

Reply:
I don't see the point of linking base R functions. We don't list `vector`, `logical`, etc.

Reply:
The term `list` is confusing with Polars list dtype, etc.

Reply:
In the docs we use "A polars ...", e.g. "A polars expression". I think it's clear enough that if we don't then we refer to standard R terms.

Reply:
Ok, that makes sense.